### PR TITLE
First stable version of the Node.js library

### PR DESCRIPTION
**Pull request by marcellobarile**
TODOs:
- more examples
- code documentation

### CODE REVIEW

**Comment by marcellobarile** [comment]
@CleoQc could you review this PR? Any feedback is appreciated :) Thanks!

File: Software/NodeJS/install.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    else
+        base="/$(basename "$f")"
+        dir=$(dirname "$f")
+    fi
+    dir=$(cd "$dir" && /bin/pwd)
+    echo "$dir$base"
+}
+
+if [ "$(id -u)" == "0" ]; then
+   echo "This script cannot be run as sudo" 1>&2
```
[the remark this answers is not in the document]
Comment (marcellobarile) [reply Software/NodeJS/install.sh]: I'll investigate on it, thanks Robert.

File: Software/NodeJS/sh/installNodejs.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+# Loading NVM
+. ~/.nvm/nvm.sh
+. ~/.profile
+. ~/.bashrc
```
[the remark this answers is not in the document]
Comment (marcellobarile) [reply Software/NodeJS/sh/installNodejs.sh]: Yes, when you install Node.js/NPM through NVM the terminal session has to be restarted.
Those 3 lines should do the trick but, maybe, there are some cases where it's not enough.

Could you please tell me the OS that you are using for your tests?

[the remark this answers is not in the document]
Comment (RobertLucian) [reply Software/NodeJS/sh/installNodejs.sh]: I used the latest Jessie image that's available in the archives.

[the remark this answers is not in the document]
Comment (marcellobarile) [reply Software/NodeJS/sh/installNodejs.sh]: I'm using Raspbian for Robots which should be based on Jessie too.
In any case I'll find a better solution for it.
Thanks again

[the remark this answers is not in the document]
Comment (RobertLucian) [reply Software/NodeJS/sh/installNodejs.sh]: Yes, that's correct. So, `Raspbian For Robots` is just an "OS" that has lots of other dependencies installed and it's based on Jessie - that's all. There's no change in how it operates.

If you want to test anything, just give me a heads-up and I'll test it.
Thanks.

**Comment by marcellobarile** [comment]
@RobertLucian @CleoQc I've applied some changes to the bash scripts and to the readme.
Please let me know if you think we should add something else to the readme, for me that's descriptive enough for someone who already has a minimum of experience w/ node.js.

Best,
Marcello

**Comment by RobertLucian** [comment]
@marcellobarile Thank you for your contribution. 

I've tested it and the install scripts now work as intended. Also, the readme looks more than clear, so that's a go from me.

**Pull request merged**